src/context.rs

```diff
@@ -1097,23 +1097,29 @@ impl BlockOffset {
 pub struct Block {
   pub mode: PredictionMode,
   pub bsize: BlockSize,
+  pub sb_type: BlockSize,
   pub partition: PartitionType,
   pub skip: bool,
   pub ref_frames: [usize; 2],
   pub neighbors_ref_counts: [usize; TOTAL_REFS_PER_FRAME],
-  pub cdef_index: u8
+  pub cdef_index: u8,
+  pub n8_w: usize, /* block width in the unit of mode_info */
+  pub n8_h: usize, /* block height in the unit of mode_info */
 }
 
 impl Block {
   pub fn default() -> Block {
     Block {
       mode: PredictionMode::DC_PRED,
       bsize: BlockSize::BLOCK_64X64,
+      sb_type: BlockSize::BLOCK_64X64,
       partition: PartitionType::PARTITION_NONE,
       skip: false,
       ref_frames: [INTRA_FRAME; 2],
       neighbors_ref_counts: [0; TOTAL_REFS_PER_FRAME],
-      cdef_index: 0
+      cdef_index: 0,
+      n8_w: BlockSize::MI_SIZE_WIDE[BLOCK_64X64 as usize],
+      n8_h: BlockSize::MI_SIZE_HIGH[BLOCK_64X64 as usize],
```

Comment:
Copying the comment I've left at atomnuker's branch.
"You probably know n8_w and n8_h means n4_w and n4_h?"
I've fixed most of these outdated names in aomenc, but missed many in aomdec."

Reply:
Yes, these names should be changed. Even n4_w and n4_h may be confusing. Can we do this refactoring in a separate PR?

Reply:
Sure- let's correct them in separate patch. There are several more defined locally.